label-application-rule-config-service-impl/src/test/java/org/hypertrace/label/application/rule/config/service/LabelApplicationRuleConfigServiceImplTest.java

```diff
@@ -44,20 +44,29 @@
 import org.junit.jupiter.api.Test;
 
 public class LabelApplicationRuleConfigServiceImplTest {
+  private static final String SYSTEM_LABEL_APPLICATION_RULE_STR =
+      "{\"id\":\"system-label-application-rule-1\",\"data\":{\"name\":\"SystemLabelApplicationRule1\",\"matching_condition\":{\"leaf_condition\":{\"key_condition\":{\"operator\":\"OPERATOR_EQUALS\",\"value\":\"test.key\"},\"unary_condition\":{\"operator\":\"OPERATOR_EXISTS\"}}},\"label_action\":{\"entity_types\":[\"API\"],\"operation\":\"OPERATION_MERGE\",\"static_labels\":{\"ids\":[\"static-label-id-1\"]}},\"enabled\":false}}";
   MockGenericConfigService mockGenericConfigService;
   LabelApplicationRuleConfigServiceBlockingStub labelApplicationRuleConfigServiceBlockingStub;
+  LabelApplicationRule systemLabelApplicationRule;
 
   @BeforeEach
-  void setUp() {
+  void setUp() throws InvalidProtocolBufferException {
     mockGenericConfigService =
         new MockGenericConfigService().mockUpsert().mockGet().mockGetAll().mockDelete();
     Channel channel = mockGenericConfigService.channel();
     ConfigChangeEventGenerator configChangeEventGenerator = mock(ConfigChangeEventGenerator.class);
-    Config config =
-        ConfigFactory.parseMap(
-            Map.of(
-                LABEL_APPLICATION_RULE_CONFIG_SERVICE_CONFIG,
-                Map.of(MAX_DYNAMIC_LABEL_APPLICATION_RULES_PER_TENANT, 2)));
+    String configStr =
```

Comment:
Likewise here - This doesn't belong in the impl test, we can mock instead and then also add a config class test that passes json and checks the config parse.

Reply:
Done